drv/sidecar-front-io-api/build.rs

```diff
@@ -0,0 +1,166 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use serde::Deserialize;
+use std::fmt::Write;
+use std::{env, fs, path::PathBuf};
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    build_util::expose_target_board();
+
+    let out_dir = &PathBuf::from(env::var_os("OUT_DIR").unwrap());
+
+    fs::write(out_dir.join("sidecar_qsfp_x32_controller.rs"), regs()?)?;
+
+    let ecp5_bitstream_name = match env::var("HUBRIS_BOARD")?.as_str() {
```

Comment:
Why are we deploying the FPGA image in `sidecar-front-io-api`, rather than the `-server` task?

e.g. on Gimlet, the image is in `gimlet-seq-server`, rather than `gimlet-seq-api`.

Reply:
After reading more, I think the crate is just confusingly named.

It doesn't feel like a `-api` crate, which is normally a thin shim around Idolatry client bindings.

How about renaming to `drv-sidecar-front-io` and `drv-sidecar-mainboard-controller`?  Then, it doesn't feel weird that it contains the FPGA image.

Reply:
Sure, we can do that. In a different world I would have called these something along the lines of `libfront_io` and `libmainboard_controller`. I can see the argument why `-api` crates are closely tied to Idol client bindings.